### DISCUSSION TITLE
E possivel usar node.js no next.js?

### DISCUSSION THREAD

**Discussion by HigorAln** [frontendbr/forum#2152] [Perguntas e Respostas]
Olá pessoal, iniciei em programação a um tempinho, e estou focado no react (nextjs) e estou indo ingressar em um curso de node js... mas minhas duvidas são...
E possivel usar node.js no next.js? e sim... e viavel usar?

**Comment by wilsonneto-swe** [comment]
O Next.js já é node, é isso que permite a ele fazer a parte SSR no server.
As rotas ./api são rotas puramente em node para api pequenoas.

Porém em uma aplicação mais robusta, gosto de deixar o next.js apenas com o front, e o back faço com express.js em uma aplicação separada.

next.js e express rodam em cima do node.js.

**Reply by HigorAln** [reply]
minha mente explodiu agora ^^, então claramente e necessário aprofundar no node pra ter mais conhecimento sobre o next.js ^^ obrigado pela resposta !!

**Reply by MechanicallyDev** [reply]
Se vocês quiserem fazer uma aplicação monolítica recomendo dar uma olhada no BlitzJS.
Ele é usa como base o NextJS, mas faz toda a abstração das rotas de API. Você importa as funções de backend direto no front e ele faz a mágica por trás dos panos.

**Comment by ninetails** [comment]
O ideal é não utilizar pacotes que forem usados só no servidor no front...

Mas tem algumas partes do Next.js que rodam apenas no servidor e que você pode usar sem medo, porém perdendo o <abbr title="Static Site Generator">SSG</abbr>: dentro do [getServerSideProps](https://nextjs.org/docs/basic-features/data-fetching#getserversideprops-server-side-rendering) por exemplo.

Mas o ideal mesmo seria abstrair este uso para o servidor (digo, API... seja a pasta API do Next.js ou uma API externa) e seu dado no front (Nextjs) vir pronto para consumo.
